bootstrap/bootstrap.go

```diff
@@ -93,26 +92,78 @@ func RunAndReturnWaitGroup(
 	var wg sync.WaitGroup
 	deferred := func() {}
 
+	lc := logging.FactoryToStdout(serviceKey)
+
+	// Enforce serviceConfig (which is an interface) is a pointer so we can dereference it later with confidence when required.
+	if reflect.TypeOf(serviceConfig).Kind() != reflect.Ptr {
```

Comment:
Don't use reflection in production code (even if it's only during initialization).  Interfaces are passed as method and function parameters by reference.  This test isn't needed.

Reply:
FYI, we are using reflection in production code in other places in this module. Reason I added this check is we use this call `reflect.ValueOf(serviceConfig).Elem().Interface()` when needing to pass the struct rather than the pointer to a other module func's. This check allows use to that with confidence that we have a pointer to deference.

Reply:
You always have a pointer.

Doing more of a thing elsewhere isn't valid justification.

If you need the underlying structure, do a type assertion.  That's what they're for.  Type assertion will return `ok` indicating whether or not the underlying type is what you expect it to be.  No need for reflection. Really.  ;)

Reply:
Yes, I verified in edgex-go that it will not compiler if the passed parameter is not a pointer. Forgot it was passed as a defined interface, not just interface{}, which always has to be a pointer.

Will remove this check.

BTW, I don't know of a type assertion syntax that would verify that a generic interface{} parameter is pointer. If you have one I would be interested for future use. Everything I found uses the reflection I have here.

Reply:
Pointer check via reflection was removed